server/conf/cms/secrets.sample.py

```diff
@@ -1,31 +1,250 @@
+# _ALLOWED_HOSTS = ['hostname.tacc.utexas.edu', 'host.ip.v4.address', '0.0.0.0', 'localhost', '127.0.0.1']   # In production.
+_ALLOWED_HOSTS = ['0.0.0.0', '127.0.0.1', 'localhost', '*']   # In development.
+
+# Boolean check to see if ldap is being used by the site.
+# Ensure the django-auth-ldap==2.0.0 package is uncommented
+# in the requirements.txt file or installed if using ldap.
+_LDAP_ENABLED = True
+
+# Boolean check to determine the appropriate database settings when using containers.
+_USING_CONTAINERS = True
+
+########################
+# DATABASE SETTINGS
+########################
+
+if _USING_CONTAINERS:
+    # used in container deployments.
+    _DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql_psycopg2',
+            'PORT': '5432',
+            'NAME': 'taccsite',
+            'USER': 'postgresadmin',
+            'PASSWORD': 'taccforever', # Change before live deployment.
+            'HOST': 'frontera_cms_postgres'
+        }
+    }
+else:
+    # used in local dev, venv or manual deployments.
+    _DATABASES = {
+        'default': {
+            'ENGINE': 'django.db.backends.postgresql_psycopg2',
+            'PORT': '5432',
+            'NAME': 'taccsite',
+            'USER': 'postgresadmin',
+            'PASSWORD': 'taccforever', # Change before live deployment.
+            'HOST': 'localhost'
+        }
+    }
+
+########################
+# DJANGO CMS SETTINGS
+########################
+
+# CMS Site (allows for multiple sites on a single CMS)
 _SITE_ID = 1
 _CMS_TEMPLATES = (
-    ('base.html', 'Base'),
-    ('cms_menu.html', 'CMS Menu'),
-    ('fullwidth.html', 'Fullwidth'),
-    ('sidebar_left.html', 'Sidebar Left'),
-    ('sidebar_right.html', 'Sidebar Right')
+    # Customize this
+    # FAQ: First template is default template
+    # REF: http://docs.django-cms.org/en/latest/how_to/install.html#templates
+    ('frontera-cms/templates/fullwidth.html', 'Fullwidth'),
+    ('fullwidth.html', 'DEPRECATED Fullwidth'),
 )
 
-# As needed:
-_CACHES = {
-    'default': {
-        'BACKEND': 'django.core.cache.backends.memcached.MemcachedCache',
-        'LOCATION': 'frontera_prtl_memcached:11211',
-    }
+########################
+# GOOGLE ANALYTICS
+########################
+
+# To use during dev, Tracking Protection in browser needs to be turned OFF.
+_GOOGLE_ANALYTICS_PROPERTY_ID = "UA-125525035-13"
+_GOOGLE_ANALYTICS_PRELOAD = True
+
+########################
+# CMS FORMS
+########################
+
+# Create CMS Forms
+# SEE: https://pypi.org/project/djangocms-forms/
+# SEE: https://www.google.com/recaptcha/admin/create
+_DJANGOCMS_FORMS_RECAPTCHA_PUBLIC_KEY = ""
+_DJANGOCMS_FORMS_RECAPTCHA_SECRET_KEY = ""
+
+########################
+# ELASTICSEARCH
+########################
+
+_ES_AUTH = 'username:password'
+_ES_HOSTS = 'http://elasticsearch:9200'
+_ES_INDEX_PREFIX = 'frontera-dev-{}'
+_ES_DOMAIN = 'http://localhost:8000'
```

Comment:
@jarosenb are these settings correct for local dev?

Reply:
I think the ES_INDEX_PREFIX is supposed to be:

_ES_INDEX_PREFIX = 'cms-dev-{}'

Reply:
Given the current state of FP-208:
- For the CMS—stand-alone—it must be `cms-dev-{}`.
- For the CMS—within the Portal—it must be `frontera-dev-{}`.

This change was required to successfully test https://github.com/TACC/Frontera-Portal/pull/276#issuecomment-745498701.

I agree that y'all should confer with @jarosenb.

Reply:
That is likely because we have not fully anonymized the frontera portal codebase setup. The name of the index is really defined when we create the index in the ES cluster, so this is likely to align with a configuration value in the es container. This change will need to be a January task most likely (culling the project-specific references from the codebase) since there is no need to complicate it now just before a deployment & break.

Reply:
Ah, yeah I see, now. I keep forgetting Frontera Portal will be a Core Portal. I should not forget this as often as I do…

I actually want to try this change, now, given that FP-208 is in progress (#276), but then [I might be a hypocrite](https://github.com/TACC/Frontera-Portal/pull/60/files#r544468925).

Reply:
I mean this IS the same file. It can have test values =).

Reply:
But if easier, we can leave for now and anonymize this all in January when we coordinate Github usage and repos and all that jazz.

Reply:
I am afraid to muck with FP-208, especially given that I have not completed my own work for it. I am stepping away.

Reply:
My understanding of this is that in the real `secrets.py` we use `_ES_INDEX_PREFIX = 'frontera-dev-{}'` because that points to the same index used by the portal container (django & web app), and right now all the search is exposed under one single index for Jake's sanity.

Because this is the example file, not a real-use situation, it should not contain any of our actual development values. This file is used if someone has no secrets yet or has not configured the portal yet, it should use a generic index as the example.

In this case, the example of `cms-dev-{}` works because in the stand alone sites and the ES index lives on the local host. 

In our portals, that would not work because the indices live on the shared ES cluster where every portal + cms requires a unique index name (which again should be defined in secrets.py, not the example file).

Reply:
I agree, and for now, this should contain whatever values are necessary to spin things up locally, simply by copying secrets.sample.py to secrets.py. In fact, I don't see the need for this sample file, since it is only ever used locally. Could this file just be `secrets.py`?

Reply:
These settings look good to me, I think the `frontera-dev-{}` prefix is appropriate here.

Reply:
Because `_ES_INDEX_PREFIX` in this cms setting should match what is in the core portal setting, we can keep this as `frontera-dev-{}` for now and tackle genericization in the new year

Reply:
@rstijerina  This file can not just be secrets.py (at least not right now) because that will break all the logic checks in our settings.py for the cms to see which file is being used. Won't devs still require setting up. secrets.py file with settings specific to the project being worked on even locally?

By ensuring the base code runs "out of the box" with the default_secrets we provide a quick check for new portal projects by providing default values, but those should be changed in an actual secrets.py file. 

@jarosenb why should the example secrets file include verbiage specific to a portal under development? No other portal will use that index, it is inappropriate as a default. If you mean because it aligns with the currently configured index in the portal container, that indicates that we need to do a better job of removing project specific naming from the codebase in general. 

I agree this can wait until next year to avoid the disruptions upstream in the portal settings for "out of the box" setup, but this kind of inconsistency spread across the codebase is unsustainable long-term and we will need to better codify the naming of things everywhere come 2021. We are just getting too big to sustain an ad-hoc approach to naming things anymore.

Reply:
[FP-808: Anonymize Frontera Portal Setting/Secret Values](https://jira.tacc.utexas.edu/browse/FP-808)